site/_posts/2021-09-16-how-to-authenticate-your-react-app.md

```diff
@@ -0,0 +1,168 @@
+
+You'll need to edit that Kickstart file and update the application name and redirect URLs.
+
+## Write the React Application
+
+We're going to use both Express and React for this application, so let's start with our folder structure. From the command line, run the following commands:
+
+Make a parent folder called `reactauthapp`:
+
+```
+mkdir reactauthapp
+```
+
+This folder will hold both the front and back end files.
+
+`cd` into that parent folder, and then run:
+```
+npx create-react-app client
+```
+This creates a React app inside the parent directory.
+
+Now, make a `server.js` file in the parent directory with:
+
+```
+touch server.js
+```
+
+This will hold our Express server.
+
+Lastly, in the `reactauthapp` folder, run:
+
+```bash
+npm init -y
+```
+Confirm that a package.json was created in the root `reactauthapp` folder.
+
+At the end, your folder structure should look like this:
+
+```
+reactauthapp
+  client
+  server.js
+  package.json
+```
+
+To cover all of your bases, I recommend `cd`-ing into your `client` folder and running
+
+```
+npm start
+```
+
+to make sure your React app starts up correctly.
+
+
+## Write the Express Server
+
+*"Wait", you may say, "why do I need an Express server? Can't I just use Router?"*
+
+Again, a good question. You need an Express server, because there needs to be somewhere for the access token to be stored that isn't on the browser to pass
+
+## Testing Your Login and Logout
+
+Visit your React site on localhost:3000 and you should see your login form with a `login` button (or a button with whatever label you’ve put on it for your site).
+
+Picture here
+
+Clicking on the login button with the correct user credentials should log you straight into the website.
+
+Picture here
+
+Clicking log out will then log you out of the site and you’ll see the login form once more.
+
+Double-check that your URLS also match up properly.
+
+More TBD
+
+## Troubleshooting
+
+FusionAuth has made troubleshooting easy. If you’ve missed a setting in your OAuth configuration, you will most likely see an error message come back telling you what you’ve missed. Check back through the settings above to make sure you have copied the correct information over from FusionAuth, and that you have saved both the OAuth settings as well as the Attribute Mapping in the miniOrange OAuth Client.
+
+You might find that after a restart or a long time between logins, your FusionAuth instance will log out. When that happens, clicking the login on your login form will add an additional step for you to log into your FusionAuth instance again so that it can access the needed credentials.
+
+Add any additional troubleshooting here
+
+## Conclusion
+
+OAuth is just one feature of FusionAuth, but you can see how easily you can set it up to make your login process more streamlined. And you can make your React apps blow everyones minds with FusionAuth.
```

Comment:
I like a 'take it further' section (explore self registration or themes or roles) or a tease for the next blog post (the iframe one).